madmom/audio/signal.py

```diff
@@ -496,13 +496,10 @@ def load_wave_file(filename, sample_rate=None, num_channels=None, start=None,
     file_sample_rate, signal = wavfile.read(filename, mmap=True)
     # if the sample rate is not the desired one, raise exception
     if sample_rate is not None and sample_rate != file_sample_rate:
-        raise ValueError('Requested sample rate of %f Hz, but got %f Hz and '
-                         're-sampling is not implemented.' %
-                         (sample_rate, file_sample_rate))
+        signal = resample(signal, sample_rate=sample_rate)
     # same for the data type
     if dtype is not None and signal.dtype != dtype:
-        raise ValueError('Requested dtype %s, but got %s and re-scaling is '
-                         'not implemented.' % (dtype, signal.dtype))
+        signal = rescale(signal, dtype=dtype)
```

Comment:
I figured it out now. My assumption was that I wouldn't ruin anything by replacing these ValueErrors by the implementation because nobody could have used that function for loading something which requires resampling or rescaling anyways. However, I was wrong... the `load_audio_file` function in signal.py first tries to load using `load_wave_file` and falls back to ffmpeg if an Error is thrown. Now, as the latter function doesn't return errors any more it loads, resamples and rescales correctly what ffmpeg did before, only that the results differ in some late decimal places. I guess that explains the failing doctests as well.

What would you do? Maybe opening a separate issue would be the right solution, as my changes turned out not to be as simple as I assumed them to be. I don't need them for my branch, so I could revert the commit.